src/KOKKOS/pair_reaxff_kokkos.cpp

```diff
@@ -199,8 +199,6 @@ void PairReaxFFKokkos<DeviceType>::setup()
   int i,j,k,m;
   int n = atom->ntypes;
 
-  setup_flag = 1;
```

Comment:
Here are the lines I was referring to: https://github.com/lammps/lammps/blob/develop/src/REAXFF/pair_reaxff.cpp#L128-L141

Reply:
@stanmoore1 well, *with* it, the destructor is trying to deallocate memory that hasn't been allocated and that leads to crashes (e.g. in the unit tests). By setting the flag without calling the allocate function of the base class, you are creating inconsistent behavior. Perhaps the KOKKOS allocate() function just needs to call the base class allocate() function?

Reply:
Is it trying to deallocate wild pointers? Maybe setting all the pointers to `nullptr` in the base class would help?

Reply:
Checked that, didn't help. Let me have a closer look again. Now with the additional input from you about why you want to deallocate stuff in the base class, there is a way.

Reply:
OK, we could call the base `allocate` as you suggested, or I can try to make it be more careful of only deallocating what is allocated.

Reply:
That doesn't seem to be the issue. It looks more like the allocations that are leaking originate from calling init_md().

Reply:
OK thanks for looking at this

Reply:
@athomps @sjplimp do you have any suggestions for how to best consolidate this?

Reply:
@akohlmey a92a3ec fixes the memory leak for my reproducer, but not sure if it will still segfault in the unit tests.

Reply:
It looks like there is a SNAFU with the N2P2 repository. The downloaded archive with the given tag has suddenly a different MD5 sum. Thus a bunch of tests seem to be failing because they fail to compile when they download and check the library.

Reply:
@akohlmey I think you mean PACE, not N2P2?

Reply:
Hmmm... seems to affect both

Reply:
ARRRGGGHHH!!!!